### DISCUSSION TITLE
How to start a dragstart event instead moving the fabtic.Object?

### DISCUSSION THREAD

**Discussion by punker76** [fabricjs/fabric.js#8921] [Q&A]
Hi,

I want to use the dragstart on a fabric.Object to allow the normal drag&drop functionality instead moving the fabric.Object itself.
Is this possible? A simple drag&drop mechanism from one object to another.

I tried to fire a dragstart event manually on the element of the canvas inside the mousemove of the canvas, but this doesn't work.

```ts
private __onMouseMove(evt: fabric.IEvent): void {
...
  // Fire a dragstart event, now that we know what to do with it
  const dataTransfer = new DataTransfer();
  // this has no effect, the property will not get the value
  dataTransfer.effectAllowed = 'move';
  console.log('DataTransfer', dataTransfer);

  var e = new DragEvent('dragstart', {
    dataTransfer: dataTransfer,
    button: evt.button,
    relatedTarget: this._canvas.getElement(),
    clientX: evt.pointer.x,
    clientY: evt.pointer.y,
  });
  console.log('DragEvent', e);
  console.log('fired', this._canvas.getElement().dispatchEvent(e));
...
}

public onDragStart(ev: DragEvent): void {
  if (ev.dataTransfer) {
    ev.dataTransfer.setData('text', 'YEAH');
    // this has no effect, the property will not get the value
    ev.dataTransfer.effectAllowed = 'move';
  }
  console.log('onDragStart', ev);
  console.log('drag data', ev.dataTransfer?.getData('Text'));
}
```

The event `onDragStart` will be fired but the dataTransfer object is not correct filled as it would be if I use the event on a div directly.

Any suggestions?

Thx!

**Comment by ShaMan123** [comment]
What are you trying to achieve?
https://github.com/fabricjs/fabric.js/blob/0f0ed963b959969e1968563b176e9e67a3d0feb5/src/shapes/IText/ITextClickBehavior.ts#L65

**Comment by punker76** [comment]
Hi, first I forgot to say that I'm using the latest v5 stable version and not the pre v6. If you say it's better to use the v6, then I will try it.

I want to achieve a drag&drop mechanism as I do it with the HTML 5 drag&drop API, so that I can work with a data and drop effects.

https://github.com/fabricjs/fabric.js/assets/658431/9a10444e-ccc9-4b6b-a903-9215adca3f68

https://web.dev/drag-and-drop/#completing-the-drop

**Reply by ShaMan123** [reply]
I implemeneted dnd for text.
The rest you will need to figure out.
Anyone asking, 5 or 6, the answer is clear. 6
